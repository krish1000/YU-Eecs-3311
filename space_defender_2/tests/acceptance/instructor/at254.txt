--Test enemy grunt actions, with and without vision.
toggle_debug_mode

play(5,12,38,38,38,38,38)
setup_select(1)
setup_next(2)
setup_select(3)
setup_next(1)
setup_select(1)
setup_next(1)
setup_next(1)

move(C,5)
pass
pass
move(C,7)
move(C,8)
