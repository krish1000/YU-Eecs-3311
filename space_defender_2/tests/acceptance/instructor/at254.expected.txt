  state:not started, normal, ok
  Welcome to Space Defender Version 2.
->toggle_debug_mode
  state:not started, debug, ok
  In debug mode.
->play(5,12,38,38,38,38,38)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_select(1)
  state:weapon setup, debug, ok
  1:Standard (A single projectile is fired in front)
    Health:10, Energy:10, Regen:0/1, Armour:0, Vision:1, Move:1, Move Cost:1,
    Projectile Damage:70, Projectile Cost:5 (energy)
  2:Spread (Three projectiles are fired in front, two going diagonal)
    Health:0, Energy:60, Regen:0/2, Armour:1, Vision:0, Move:0, Move Cost:2,
    Projectile Damage:50, Projectile Cost:10 (energy)
  3:Snipe (Fast and high damage projectile, but only travels via teleporting)
    Health:0, Energy:100, Regen:0/5, Armour:0, Vision:10, Move:3, Move Cost:0,
    Projectile Damage:1000, Projectile Cost:20 (energy)
  4:Rocket (Two projectiles appear behind to the sides of the Starfighter and accelerates)
    Health:10, Energy:0, Regen:10/0, Armour:2, Vision:2, Move:0, Move Cost:3,
    Projectile Damage:100, Projectile Cost:10 (health)
  5:Splitter (A single mine projectile is placed in front of the Starfighter)
    Health:0, Energy:100, Regen:0/10, Armour:0, Vision:0, Move:0, Move Cost:5,
    Projectile Damage:150, Projectile Cost:70 (energy)
  Weapon Selected:Standard
->setup_next(2)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Standard
->setup_select(3)
  state:engine setup, debug, ok
  1:Standard
    Health:10, Energy:60, Regen:0/2, Armour:1, Vision:12, Move:8, Move Cost:2
  2:Light
    Health:0, Energy:30, Regen:0/1, Armour:0, Vision:15, Move:10, Move Cost:1
  3:Armoured
    Health:50, Energy:100, Regen:0/3, Armour:3, Vision:6, Move:4, Move Cost:5
  Engine Selected:Armoured
->setup_next(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_select(1)
  state:power setup, debug, ok
  1:Recall (50 energy): Teleport back to spawn.
  2:Repair (50 energy): Gain 50 health, can go over max health. Health regen will not be in effect if over cap.
  3:Overcharge (up to 50 health): Gain 2*health spent energy, can go over max energy. Energy regen will not be in effect if over cap.
  4:Deploy Drones (100 energy): Clear all projectiles.
  5:Orbital Strike (100 energy): Deal 100 damage to all enemies, affected by armour.
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:setup summary, debug, ok
  Weapon Selected:Standard
  Armour Selected:None
  Engine Selected:Armoured
  Power Selected:Recall (50 energy): Teleport back to spawn.
->setup_next(1)
  state:in game(0.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:110/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,1]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
  Enemy Action:
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _
    C S  _  _  _  _  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->move(C,5)
  state:in game(1.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:86/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,5]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,12]
  Projectile:
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) moves: [C,1] -> [C,5]
  Enemy Action:
  Natural Enemy Spawn:
    A Grunt(id:1) spawns at location [A,12].
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  _  _  _  G
    B _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(2.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:94/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,5]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
    [2,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,12]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,9]
  Friendly Projectile Action:
  Enemy Projectile Action:
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,5], doubling regen rate.
  Enemy Action:
    A Grunt(id:1) gains 10 total health.
    A Grunt(id:1) moves: [A,12] -> [A,10]
      A enemy projectile(id:-1) spawns at location [A,9].
  Natural Enemy Spawn:
    A Grunt(id:2) spawns at location [A,12].
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  _  _  _  _  <  G  _  G
    B _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->pass
  state:in game(3.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:102/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,5]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:120/120, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,8]
    [2,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,10]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,5]
    [-2,<]->damage:15, move:4, location:[A,7]
    [-3,<]->damage:15, move:4, location:[A,9]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,9] -> [A,5]
  Starfighter Action:
    The Starfighter(id:0) passes at location [C,5], doubling regen rate.
  Enemy Action:
    A Grunt(id:1) gains 10 total health.
    A Grunt(id:2) gains 10 total health.
    A Grunt(id:1) moves: [A,10] -> [A,8]
      A enemy projectile(id:-2) spawns at location [A,7].
    A Grunt(id:2) moves: [A,12] -> [A,10]
      A enemy projectile(id:-3) spawns at location [A,9].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12
    A _  _  _  _  <  _  <  G  <  G  _  _
    B _  _  _  _  _  _  _  _  _  _  _  _
    C _  _  _  _  S  _  _  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->move(C,7)
  state:in game(4.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:94/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,7]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:120/120, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,4]
    [2,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[A,6]
    [3,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[B,12]
  Projectile:
    [-1,<]->damage:15, move:4, location:[A,1]
    [-4,<]->damage:30, move:4, location:[A,3]
    [-5,<]->damage:15, move:4, location:[A,5]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,5] -> [A,1]
    A enemy projectile(id:-2) moves: [A,7] -> [A,3]
    A enemy projectile(id:-3) moves: [A,9] -> [A,8]
      The projectile collides with Grunt(id:1) at location [A,8], healing 15 damage.
  Starfighter Action:
    The Starfighter(id:0) moves: [C,5] -> [C,7]
  Enemy Action:
    A Grunt(id:1) moves: [A,8] -> [A,4]
      A enemy projectile(id:-4) spawns at location [A,3].
      The projectile collides with enemy projectile(id:-2) at location [A,3], combining damage.
    A Grunt(id:2) moves: [A,10] -> [A,6]
      A enemy projectile(id:-5) spawns at location [A,5].
  Natural Enemy Spawn:
    A Grunt(id:3) spawns at location [B,12].
      1  2  3  4  5  6  7  8  9 10 11 12
    A <  _  <  G  <  G  _  _  _  _  _  _
    B _  _  _  _  _  _  _  _  _  _  _  G
    C _  _  _  _  _  _  S  _  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
->move(C,8)
  state:in game(5.0), debug, ok
  Starfighter:
    [0,S]->health:110/110, energy:92/110, Regen:1/4, Armour:3, Vision:7, Move:6, Move Cost:6, location:[C,8]
      Projectile Pattern:Standard, Projectile Damage:70, Projectile Cost:5 (energy)
      Power:Recall (50 energy): Teleport back to spawn.
      score:0
  Enemy:
    [1,G]->health:120/120, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:F, can_see_Starfighter:F, location:[A,2]
    [2,G]->health:110/110, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:F, location:[A,3]
    [3,G]->health:100/100, Regen:1, Armour:1, Vision:5, seen_by_Starfighter:T, can_see_Starfighter:T, location:[B,8]
  Projectile:
    [-6,<]->damage:15, move:4, location:[A,1]
    [-8,<]->damage:15, move:4, location:[B,7]
  Friendly Projectile Action:
  Enemy Projectile Action:
    A enemy projectile(id:-1) moves: [A,1] -> out of board
    A enemy projectile(id:-4) moves: [A,3] -> out of board
    A enemy projectile(id:-5) moves: [A,5] -> [A,4]
      The projectile collides with Grunt(id:1) at location [A,4], healing 15 damage.
  Starfighter Action:
    The Starfighter(id:0) moves: [C,7] -> [C,8]
  Enemy Action:
    A Grunt(id:1) moves: [A,4] -> [A,2]
      A enemy projectile(id:-6) spawns at location [A,1].
    A Grunt(id:2) moves: [A,6] -> [A,3]
      A enemy projectile(id:-7) spawns at location [A,2].
      The projectile collides with Grunt(id:1) at location [A,2], healing 15 damage.
    A Grunt(id:3) moves: [B,12] -> [B,8]
      A enemy projectile(id:-8) spawns at location [B,7].
  Natural Enemy Spawn:
      1  2  3  4  5  6  7  8  9 10 11 12
    A <  G  G  _  _  _  _  _  _  _  _  _
    B _  _  _  _  _  _  <  G  _  _  _  _
    C _  _  _  _  _  _  _  S  _  _  _  _
    D _  _  _  _  _  _  _  _  _  _  _  _
    E _  _  _  _  _  _  _  _  _  _  _  _
